Mathlib/Analysis/SpecialFunctions/PolarCoord.lean

```diff
@@ -121,29 +129,36 @@ theorem polarCoord_source_ae_eq_univ : polarCoord.source =ᵐ[volume] univ := by
 theorem integral_comp_polarCoord_symm {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
     (f : ℝ × ℝ → E) :
     (∫ p in polarCoord.target, p.1 • f (polarCoord.symm p)) = ∫ p, f p := by
-  set B : ℝ × ℝ → ℝ × ℝ →L[ℝ] ℝ × ℝ := fun p =>
-    LinearMap.toContinuousLinearMap (Matrix.toLin (Basis.finTwoProd ℝ) (Basis.finTwoProd ℝ)
-      !![cos p.2, -p.1 * sin p.2; sin p.2, p.1 * cos p.2])
-  have A : ∀ p ∈ polarCoord.symm.source, HasFDerivAt polarCoord.symm (B p) p := fun p _ =>
-    hasFDerivAt_polarCoord_symm p
-  have B_det : ∀ p, (B p).det = p.1 := by
-    intro p
-    conv_rhs => rw [← one_mul p.1, ← cos_sq_add_sin_sq p.2]
-    simp only [B, neg_mul, LinearMap.det_toContinuousLinearMap, LinearMap.det_toLin,
-      Matrix.det_fin_two_of, sub_neg_eq_add]
-    ring
   symm
   calc
     ∫ p, f p = ∫ p in polarCoord.source, f p := by
       rw [← setIntegral_univ]
       apply setIntegral_congr_set
       exact polarCoord_source_ae_eq_univ.symm
-    _ = ∫ p in polarCoord.target, abs (B p).det • f (polarCoord.symm p) := by
-      apply integral_target_eq_integral_abs_det_fderiv_smul volume A
+    _ = ∫ p in polarCoord.target, |p.1| • f (polarCoord.symm p) := by
+      rw [← PartialHomeomorph.symm_target, integral_target_eq_integral_abs_det_fderiv_smul volume
+      (fun p _ ↦ hasFDerivAt_polarCoord_symm p), PartialHomeomorph.symm_source]
+      simp_rw [FDerivAt_polarCoord_symm_det]
     _ = ∫ p in polarCoord.target, p.1 • f (polarCoord.symm p) := by
       apply setIntegral_congr_fun polarCoord.open_target.measurableSet fun x hx => ?_
-      rw [B_det, abs_of_pos]
-      exact hx.1
+      rw [abs_of_pos hx.1]
+
+theorem lintegral_comp_polarCoord_symm (f : ℝ × ℝ → ENNReal) :
+    ∫⁻ (p : ℝ × ℝ) in polarCoord.target, (p.1).toNNReal • f (polarCoord.symm p) =
```

Comment:
```suggestion
    ∫⁻ (p : ℝ × ℝ) in polarCoord.target, ENNReal.ofReal p.1 • f (polarCoord.symm p) =
```
Since you're working with ennreals, it seems more natural to put directly the function from reals to ennreals, instead of going through an additional coercion.

Reply:
Not sure you've done this throughout the whole PR yet?

Reply:
Yeah, I am sure. I was having git troubles and got distracted. I'll fix that.